core/chain-configs/src/genesis_config.rs

```diff
@@ -234,20 +234,54 @@ impl From<&GenesisConfig> for AllEpochConfig {
 )]
 pub struct GenesisRecords(pub Vec<StateRecord>);
 
+/// custom deserializer that does the same thing that #[serde(default)] would do --
+/// if no value is provided in json, returns default value.
+fn no_value_as_default<'de, D, T>(de: D) -> Result<T, D::Error>
+where
+    D: Deserializer<'de>,
+    T: Deserialize<'de> + Default,
+{
+    let opt = Option::<T>::deserialize(de)?;
+    match opt {
+        None => Ok(T::default()),
+        Some(value) => Ok(value),
+    }
+}
+
+#[derive(Debug, Clone, SmartDefault, serde::Serialize, serde::Deserialize)]
+#[serde(untagged)]
```

Comment:
I am unsure about this, but I think you wouldn't avoid obscure serde annotations because they are inescapable part of any custom serialiser/deserialiser + you will have hardcode that is specific to one implementation of one struct. Kinda combines the worst of both approaches.
But I agree that other solutions might exist. I will request a bit more critique from rust stream on Zulip and if none comes by EOD, I will keep this solution and merge.

Reply:
It's not the first instance of our use of `serde(untagged)`.
Yes, it's confusing, but that is fine, we have examples as documentation.

Reply:
Hehe, I can't read, thought the comment was about another line. Please, disregard my previous comment. 
I will make make an explicit comment to look at tests to understand the resulting file structure.